### PR TITLE
Keyboard shortcuts - Admin Settings

### PR DESCRIPTION
**Pull request by jacob-nv**
## Description

Adding keyboard navigation to admin settings.

Fixes https://github.com/owncloud/web/issues/9481

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests

## Checklist:

- [x] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

File: packages/web-app-admin-settings/src/composables/scrollTo/useScrollTo.ts [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    const resource = unref(resources).find((r) => r.id === unref(scrollTo))
+    if (resource) {
+      store.commit('Files/SET_FILE_SELECTION', [resource])
+      scrollToResource(resource.id, { forceScroll: true })
+
+      if (unref(details)) {
+        eventBus.publish(SideBarEventTopics.openWithPanel, unref(details))
+      }
+    }
+  }
```
Comment (JammingBen) [inline packages/web-app-admin-settings/src/composables/scrollTo/useScrollTo.ts:61]: I think we can remove this, it's heavily bound to the files app (resource usage, store commit on `Files/SET_FILE_SELECTION`).

**Review by JammingBen — CHANGES_REQUESTED** [review]
Shift + click is a bit buggy for me. It only works the first time, but after de-selecting all resources and trying again, it won't.

Also, could you please make a single click on the row work like in the files app? So you don't necessarily need to click in the checkbox directly.

**Comment by JammingBen** [comment]
Still facing a few issues:

* Clicking the "(de-)select all" checkbox on the top left side breaks shift+click selection
* Clicking a single row (mouse only, no shift or ctrl) should de-select previously selected rows (see files app)
* Select all users via the "(de-)select all" checkbox, then click on one of the selected users -> it will disappear from the list

**Comment by jacob-nv** [comment]
> Still facing a few issues:
> 
> * Clicking the "(de-)select all" checkbox on the top left side breaks shift+click selection
> * Clicking a single row (mouse only, no shift or ctrl) should de-select previously selected rows (see files app)
> * Select all users via the "(de-)select all" checkbox, then click on one of the selected users -> it will disappear from the list

Everything fixed

File: packages/web-app-admin-settings/src/composables/keyboardActions/useKeyboardTableMouseActions.ts [hunk 2]
```diff
@@ -0,0 +1,77 @@
+import { onBeforeUnmount, onMounted, unref, Ref } from 'vue'
+import { eventBus } from 'web-pkg'
+import { KeyboardActions } from 'web-pkg/src/composables/keyboardActions'
+import { findIndex, find } from 'lodash-es'
+import { Resource } from 'web-client'
+
+export const useKeyboardTableMouseActions = (
+  keyActions: KeyboardActions,
+  paginatedResources: Ref<Resource[]>,
```
Comment (JammingBen) [inline packages/web-app-admin-settings/src/composables/keyboardActions/useKeyboardTableMouseActions.ts:9]: `Resource` is only for files, we need `Group[] | User[] | Space[]` or maybe a generic interface? We probably only need the `id` prop, right?

File: packages/web-app-admin-settings/src/composables/keyboardActions/useKeyboardTableMouseActions.ts [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+  selectedRows: any,
+  lastSelectedRowIndex: Ref<number>,
+  lastSelectedRowId: Ref<string | null>
+) => {
+  let fileListClickedMetaEvent
+  let fileListClickedShiftEvent
```
Comment (JammingBen) [inline packages/web-app-admin-settings/src/composables/keyboardActions/useKeyboardTableMouseActions.ts:15]: `fileList...` -> `resourceList...` I propose.

File: packages/web-app-admin-settings/src/composables/keyboardActions/useKeyboardTableMouseActions.ts [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+
+  onMounted(() => {
+    fileListClickedMetaEvent = eventBus.subscribe(
+      'app.files.list.clicked.meta',
+      handleCtrlClickAction
+    )
+    fileListClickedShiftEvent = eventBus.subscribe(
+      'app.files.list.clicked.shift',
+      handleShiftClickAction
+    )
```
Comment (JammingBen) [inline packages/web-app-admin-settings/src/composables/keyboardActions/useKeyboardTableMouseActions.ts:70]: File is not fitting here, I'd replace it with `resource` again.

**Review by JammingBen — COMMENTED** [review]
Works well! Just some nitpicks regarding terminology & types from my side.

Edit: Ah, unit tests are still failing: https://drone.owncloud.com/owncloud/web/37984/9/5

**Pull request merged**